### PR TITLE
ZEPPELIN-3983. Travis fails due to downloading spark takes a lot of time

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
This PR did several refactoring.
* rename SparkDownloadUtils -> DownloadUtils as it is also used for downloading flink.
* try apache mirror just one time and fall back to apache archive when fails to download from mirror site.  

### What type of PR is it?
[Refactoring]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://jira.apache.org/jira/browse/ZEPPELIN-3983

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@felixcheung Thanks for review, I did more refactoring

File: zeppelin-interpreter-integration/src/main/test/org/apache/zeppelin/integration/DownloadUtils.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        long startTime = System.currentTimeMillis();
+        while ((line = br.readLine()) != null) {
+          // logging per 5 seconds
+          if ((System.currentTimeMillis() - startTime) > 5000) {
+            LOGGER.info(line);
+            startTime = System.currentTimeMillis();
+          }
+        }
+      } catch (IOException ioe) {
+        ioe.printStackTrace();
```
Comment (felixcheung) [inline zeppelin-interpreter-integration/src/main/test/org/apache/zeppelin/integration/DownloadUtils.java]: LOGGER.warn()?

**Comment by felixcheung** [comment]
LGTM

**Comment by Leemoonsoo** [comment]
Looks great! +1

File: zeppelin-interpreter-integration/src/main/test/org/apache/zeppelin/integration/DownloadUtils.java (hunk 1 above)
Comment (zjffdu) [reply zeppelin-interpreter-integration/src/main/test/org/apache/zeppelin/integration/DownloadUtils.java]: Fixed

**Pull request closed without merging**